pkg/status/status.go

```diff
@@ -200,3 +227,71 @@ func (s *status) updateStatus() {
 		log.Fatalf("Error %v updating ClusterOperator", err)
 	}
 }
+
+// syncChannelConsumer will listen on the sync channel and update the status'
+// syncs and successfulSyncs fields until the stopCh is closed.
+func syncChannelConsumer(instance *status, stopCh <-chan struct{}) {
+	log.Info("Starting sync consumer")
+	go wait.Until(func() {
+		// Listen on SyncCh and update instance accordingly
+		err := <-syncCh
+		instance.syncs++
+		if err == nil {
+			instance.successfulSyncs++
+		}
+		log.Infof("Successful Syncs / Total Syncs: %d/%d", instance.successfulSyncs, instance.syncs)
+
+	}, 0*time.Second, stopCh)
+}
+
+// monitorClusterStatus updates the clusteroperator's status based on
+// the number of successful syncs / total syncs
+func (s *status) monitorClusterStatus(stopCh <-chan struct{}) {
+	// exit if cluster operator api is not present
+	if s.coAPINotPresent {
+		return
+	}
+
+	go syncChannelConsumer(instance, stopCh)
+	go wait.Until(func() {
+		// create the cluster operator in an initial state if it does not exist
+		if s.clusterOperator == nil {
+			s.setProgressing(fmt.Sprintf("Progressing towards %s", s.version))
```

Comment:
So this will happen only on a fresh cluster. Don't we need to show progressing on upgrades?

Reply:
We do need to show progressing on upgrades.

This code will show progressing on upgrades.The `instance` variable is created when the operator starts the `status.StartReporting` method without a `clusterOperator` and this pointer isn't updated to reflect what is in on cluster until `s.setProgressing` is called.